docs: update config management plugin docs (#11690)
argoproj/argo-cd: docs/user-guide/config-management-plugins.md

> There are two ways to install a Config Management Plugin:

Reviewer: ```suggestion
From v2.5 and going forward there are two ways to install a Config Management Plugin:
```

> 1. Add the plugin as a sidecar to the repo-server Pod.
>    This is a good option for a more complex plugin that would clutter the Argo CD ConfigMap. A copy of the repository is
>    sent to the sidecar container as a tarball and processed individually per application, which makes it a good option
>    for [concurrent processing of monorepos](../operator-manual/high_availability.md#enable-concurrent-processing).
> 2. Add the plugin config to the Argo CD ConfigMap (**this method is deprecated and will be removed in a future 
>    version**). The repo-server container will run your plugin's commands. This is a good option for a simple plugin that
>    requires only a few lines of code that fit nicely in the Argo CD ConfigMap.

Reviewer: Don't rely on numbers, name the bullets and then use the same names for the section titles below.

Use `a plugin` not `the plugin` as users can have multiple (I have two) -- and it's quite possible for them to migrate incrementally (I'd recommend that...).

```suggestion
* Sidecar plugin
  Add a plugin as a sidecar to the repo-server Pod.
   This is recommended. A copy of the repository is
   sent to the sidecar container as a tarball and processed individually per application, which makes it a good option
   for [concurrent processing of monorepos](../operator-manual/high_availability.md#enable-concurrent-processing).
* ConfigMap plugin
  (Deprecated)
  Add a plugin config to the Argo CD ConfigMap (**this method is deprecated and will be removed in a future 
   version**). The repo-server container will run your plugin's commands. This is a good option for a simple plugin that
   requires only a few lines of code that fit nicely in the Argo CD ConfigMap.
   
```

> ### Option 1: Configure plugin via sidecar

Reviewer: ```suggestion
### Sidecar plugin 
```

    2. Make sure that sidecar container is running as user 999.
    3. Make sure that plugin configuration file is present at `/home/argocd/cmp-server/config/plugin.yaml`. It can either be volume mapped via configmap or baked into image.

> ### Option 2: Configure plugins via Argo CD configmap (deprecated)

Reviewer: ```suggestion
### ConfigMap plugin
```

    The `lockRepo` key is not relevant for sidecar plugins, because sidecar plugins do not share a single source repo
    directory when generating manifests.

> ### 2. Write discovery rules for your plugin, or use the plugin name

Reviewer: As noted above, don't include numbers or periods or colons in section headings in general as they're really painful for link / search purposes.

Use very short text items so that users can scan/click/etc.
